identity-account-storage/src/storage/memstore.rs

```diff
@@ -124,51 +108,64 @@ impl Storage for MemStore {
         let public: x25519::PublicKey = secret.public_key();
         let public_key: PublicKey = public.to_bytes().to_vec().into();
 
-        let keypair: KeyPair = KeyPair::from((KeyType::X25519, public_key.clone(), private_key));
-        vault.insert(location.clone(), keypair);
+        let keypair: KeyPair = KeyPair::from((KeyType::X25519, public_key, private_key));
+        vault.insert(location.to_owned(), keypair);
 
-        Ok(public_key)
+        Ok(())
       }
-      MethodType::MerkleKeyCollection2021 => {
-        todo!("[MemStore::key_insert] Handle MerkleKeyCollection2021")
+    }
+  }
+
+  async fn key_move(&self, account_id: &AccountId, source: &KeyLocation, target: &KeyLocation) -> Result<()> {
+    let mut vaults: RwLockWriteGuard<'_, _> = self.vaults.write()?;
+
+    if let Some(vault) = vaults.get_mut(account_id) {
+      match vault.remove(source) {
+        Some(key) => {
+          vault.insert(target.to_owned(), key);
+          Ok(())
+        }
+        None => Err(Error::KeyNotFound),
       }
+    } else {
+      Err(Error::KeyVaultNotFound)
     }
   }
 
-  async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool> {
+  async fn key_exists(&self, account_id: &AccountId, location: &KeyLocation) -> Result<bool> {
     let vaults: RwLockReadGuard<'_, _> = self.vaults.read()?;
 
-    if let Some(vault) = vaults.get(did) {
+    if let Some(vault) = vaults.get(account_id) {
       return Ok(vault.contains_key(location));
     }
 
     Ok(false)
   }
 
-  async fn key_get(&self, did: &IotaDID, location: &KeyLocation) -> Result<PublicKey> {
+  async fn key_public(&self, account_id: &AccountId, location: &KeyLocation) -> Result<PublicKey> {
     let vaults: RwLockReadGuard<'_, _> = self.vaults.read()?;
-    let vault: &MemVault = vaults.get(did).ok_or(Error::KeyVaultNotFound)?;
+    let vault: &MemVault = vaults.get(account_id).ok_or(Error::KeyVaultNotFound)?;
     let keypair: &KeyPair = vault.get(location).ok_or(Error::KeyNotFound)?;
 
     Ok(keypair.public().clone())
   }
 
-  async fn key_del(&self, did: &IotaDID, location: &KeyLocation) -> Result<()> {
+  async fn key_del(&self, account_id: &AccountId, location: &KeyLocation) -> Result<()> {
     let mut vaults: RwLockWriteGuard<'_, _> = self.vaults.write()?;
-    let vault: &mut MemVault = vaults.get_mut(did).ok_or(Error::KeyVaultNotFound)?;
+    let vault: &mut MemVault = vaults.get_mut(account_id).ok_or(Error::KeyVaultNotFound)?;
 
     vault.remove(location);
 
     Ok(())
   }
```

Comment:
It's often useful to know if a command like this did anything, even if it's idempotent. Maybe return a `bool` if possible?